pkg/kots/rego/builders-opa.rego

```diff
@@ -0,0 +1,94 @@
+package kots.spec.builders
+
+# Files set with the contents of each file as json
+files[output] {
+  file := input[_]
+  output := {
+    "name": file.name,
+    "path": file.path,
+    "content": yaml.unmarshal(file.content),
+    "docIndex": object.get(file, "docIndex", 0)
+  }
+}
+
+# Check if Preflight spec exists
+v1beta1_preflight_spec_exists {
+  file := files[_]
+  file.content.kind == "Preflight"
+  file.content.apiVersion == "troubleshoot.replicated.com/v1beta1"
+}
+v1beta2_preflight_spec_exists {
+  file := files[_]
+  file.content.kind == "Preflight"
+  file.content.apiVersion == "troubleshoot.sh/v1beta2"
+}
+lint[output] {
+  rule_name := "preflight-spec"
+  rule_config := lint_rule_config(rule_name, "warn")
+  not rule_config.off
+  not v1beta1_preflight_spec_exists
+  not v1beta2_preflight_spec_exists
+  output := {
+    "rule": rule_name,
+    "type": rule_config.level,
+    "message": "Missing preflight spec"
+  }
+}
+
+# Check if informer labels exist
+wanted_informer_labels = [
+  "app.kubernetes.io/managed-by",
+  "app.kubernetes.io/name",
+  "app.kubernetes.io/instance"
+]
+
+informer_labels_exists {
+  file := files[_]
+  wanted_label := wanted_informer_labels[_]
+  file.content.metadata.labels[wanted_label]
```

Comment:
i would rename the rule to be something like `informer_labels_missing` and change this line to:
```suggestion
  not file.content.metadata.labels[wanted_label]
```

then remove the `not` in the lint rule below.

Reply:
I went with set implementation.  The suggested change failed when one file passed and one failed. In this case we want the check to pass. In this iteration. This may change in the future.